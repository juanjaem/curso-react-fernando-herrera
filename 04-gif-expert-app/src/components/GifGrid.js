import React, { Fragment } from 'react';
import PropTypes from 'prop-types';
import { useFectchGifs } from '../hooks/useFetchGifs';
import { GifGridItem } from './GifGridItem';

export const GifGrid = ({ category }) => {
    const { data: images, loading } = useFectchGifs(category);

    return (
        <Fragment>
            <h3 className="animate__animated animate__fadeIn">{category}</h3>

            { loading && <p className="animate__animated animate__flash">Cargando...</p> }

            <div className="card-grid">
                {
                    images.map(img => (<GifGridItem key={ img.id } {...img} />))
                }
            </div>
        </Fragment>
    )
}

GifGrid.propTypes = {
    category: PropTypes.string.isRequired
}
