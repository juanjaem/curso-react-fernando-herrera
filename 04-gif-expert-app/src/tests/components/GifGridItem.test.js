import { shallow } from 'enzyme';
import React from 'react';
import { GifGridItem } from '../../components/GifGridItem';

describe('pruebas en <GifGridItem />', () => {
    const title = 'Un t√≠tulo';
    const url = 'https://www.loquesea.com';
    const wrapper = shallow(<GifGridItem title={title} url={url} />)

    test('debe de mostrar el componente correctamente', () => {
        expect(wrapper).toMatchSnapshot();
    });

    test('debe tener un parrafo con el title', () => {
        const p = wrapper.find('p');
        expect(p.text().trim()).toBe(title);
    });

    test('debe tener una imagen con url y alt de los props', () => {
        const img = wrapper.find('img');
        expect(img.prop('src')).toBe(url);
        expect(img.prop('alt')).toBe(title);
    });

    test('debe de tener la clase animate__fadeIn', () => {
        const div = wrapper.find('div');
        expect(div.prop('className')).toContain('animate__fadeIn')
    });
})
