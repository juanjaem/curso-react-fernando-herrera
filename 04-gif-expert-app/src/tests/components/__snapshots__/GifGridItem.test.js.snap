// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pruebas en <GifGridItem /> debe de mostrar el componente correctamente 1`] = `
<div
  className="card animate__animated animate__fadeIn"
>
  <img
    alt="Un título"
    src="https://www.loquesea.com"
  />
  <p>
    Un título
  </p>
</div>
`;
